src/sftp.rs

```diff
@@ -474,7 +479,7 @@ impl<'sftp> Read for File<'sftp> {
             let rc =
                 raw::libssh2_sftp_read(self.raw, buf.as_mut_ptr() as *mut _, buf.len() as size_t);
             match rc {
-                n if n < 0 => Err(io::Error::new(ErrorKind::Other, self.sftp.last_error())),
+                n if n < 0 =>  Err(Error::from_errno(n as i32).into()),
```

Comment:
`n as i32` should be `n as libc::c_int` for best portability, however, I would probably just write `n as _` here and let the compiler figure it out.

Reply:
73319cf